In [43]:
from math import sqrt
from scipy import stats

# Домашняя работа по уроку 4.  Непрерывная случайная величина

-----

## Задача 1

Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].

Найдите ее среднее значение и дисперсию.

### Решение

Для равномерного распределения используются формулы:

Среднее значение - М(х) = (а + b) / 2

Дисперсия - D = ((b - a)^2) / 12

In [44]:
# M(x)
a = 200
b = 800
m = (a + b) / 2
m

500.0

In [45]:
# D
d = (b - a) ** 2 / 12 
d

30000.0

Ответ: 500; 30000

------

## Задача 2

О случайной непрерывной равномерно распределенной величине B известно, 

что ее дисперсия равна 0.2.

Можно ли найти правую границу величины B и ее среднее значение зная, 

что левая граница равна 0.5?

Если да, найдите ее.

### Решение

Можно:

D = 0.2

a = 0.5

Для равномерного распределения используются формулы:

Среднее значение - М(х) = (а + b) / 2

Дисперсия - D = ((b - a)^2) / 12

0.2 = ((b - 0.5)^2) / 12

2.4 = (b - 0.5)^2

In [46]:
# b
b = sqrt(2.4) + 0.5
b

2.049193338482967

In [47]:
# M(x)
a = 0.5
m = (a + b) / 2 
m

1.2745966692414834

Ответ: b=2.04919; M(x)=1.2746

-----------

## Задача 3

Непрерывная случайная величина X распределена нормально

и задана плотностью распределения

f(x) = (1 / (4 * sqrt(2pi))) * exp((-(x+2)**2) / 32)

Найдите:

а). M(X)

б). D(X)

в). std(X) (среднее квадратичное отклонение)

### Решение

Общий вид формулы плотности распределения нормально распределенной СВ Х имеет вид

f(x) = (1 / (std(X) * sqrt(2pi))) * exp((-(x-M(X))^2) / (2*D(X)))

Отсюда, можно заметить, что

а) M(X) = -2

б) D(X) = 32 / 2 = 16

в) std(X) = 4

Ответ: -2; 16; 4

------------

## Задача 4

Рост взрослого населения города X имеет нормальное распределение, 

причем, средний рост равен 174 см, а среднее квадратическое отклонение равно 8 см. 

Посчитайте, какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:

1. больше 182 см?

2. больше 190 см?

3. от 166 см до 190 см?

4. от 166 см до 182 см?

5. от 158 см до 190 см?

6. не выше 150 см или не ниже 190 см?

7. не выше 150 см или не ниже 198 см?

8. ниже 166 см?

Задачу можно решить двумя способами: 

без использования сторонних библиотек (numpy, scipy, pandas и пр.), 

а затем проверить себя с помощью встроенных функций

### Решение

Для решения данной задачи будем использовать z-таблицу

и формулу для нахождения Z-значений

Z = (X - mu) / std(X)

mu = 174, std(X) = 8

In [48]:
mu = 174
std = 8

def z_val(x):
    return (x - mu) / std

In [49]:
# 1. больше 182 см? - P(x > 182) = 1 - P(x < 182) (т.к. в z-таблице мы можем искать доли для значений только ниже указанного х)
#            ///////////
# ----------|----------
#          182
z_val(182)

1.0

In [50]:
# P(x < 182) = .84134 (для 1.0 в z-таблице)
# P(x > 182) = 1 - .84134
1 - .84134

0.15866000000000002

In [51]:
# используя библиотеку scipy
1 - stats.norm.cdf(182, loc=174, scale=8)

0.15865525393145707

In [52]:
# 2. больше 190 см? - P(x > 190) = 1 - P(x < 190) (т.к. в z-таблице мы можем искать доли для значений только ниже указанного х)
#            ///////////
# ----------|----------
#          190
z_val(190)

2.0

In [53]:
# P(x < 190) = .97725 (для 2.0 в z-таблице)
# P(x > 190) = 1 - .97725
1 - .97725

0.022750000000000048

In [54]:
# используя библиотеку scipy
1 - stats.norm.cdf(190, loc=174, scale=8)

0.02275013194817921

In [55]:
# 3. от 166 см до 190 см? - P(166 < x < 190) = P(x < 190) - P(x < 166) = F(190) - F(166) 
#            ///////////
# ----------|-----------|----------
#          166         190
z_val(166)

-1.0

In [56]:
# P(x < 166) = .15866 (для -1.0 в z-таблице)
# P(166 < x < 190) = .97725 - .15866
.97725 - .15866

0.8185899999999999

In [57]:
# используя библиотеку scipy
stats.norm.cdf(190, loc=174, scale=8) - stats.norm.cdf(166, loc=174, scale=8)

0.8185946141203637

In [58]:
# 4. от 166 см до 182 см? - P(166 < x < 182) = P(x < 182) - P(x < 166) = F(182) - F(166) 
#            ///////////
# ----------|-----------|----------
#          166         182
# P(x < 182) = .84134 (для 1.0 в z-таблице)
# P(x < 166) = .15866 (для -1.0 в z-таблице)
# P(166 < x < 182) = .84134 - .15866
.84134 - .15866

0.68268

In [59]:
# используя библиотеку scipy
stats.norm.cdf(182, loc=174, scale=8) - stats.norm.cdf(166, loc=174, scale=8)

0.6826894921370859

In [60]:
# 5. от 158 см до 190 см? - P(158 < x < 190) = P(x < 190) - P(x < 158) = F(190) - F(158) 
#            ///////////
# ----------|-----------|----------
#          158         190
z_val(158)

-2.0

In [61]:
# P(x < 158) = .02275 (для -2.0 в z-таблице)
# P(158 < x < 190) = .97725 - .02275
.97725 - .02275

0.9544999999999999

In [62]:
# используя библиотеку scipy
stats.norm.cdf(190, loc=174, scale=8) - stats.norm.cdf(158, loc=174, scale=8)

0.9544997361036416

In [63]:
# 6. не выше 150 см или не ниже 190 см? - P(x < 150) + (1 - P(x < 190)) 
# ///////////            ///////////
# ----------|-----------|----------
#          150         190
z_val(150)

-3.0

In [64]:
# P(x < 190) = .97725 (для 2.0 в z-таблице)
# P(x < 150) = .00135 (для -3.0 в z-таблице)
.00135 + 1 - .97725

0.02410000000000001

In [65]:
# используя библиотеку scipy
stats.norm.cdf(150, loc=174, scale=8) + 1 - stats.norm.cdf(190, loc=174, scale=8)

0.0241000299798092

In [66]:
# 7. не выше 150 см или не ниже 198 см? - P(x < 150) + (1 - P(x < 198)) 
# ///////////            ///////////
# ----------|-----------|----------
#          150         198
z_val(198)

3.0

In [67]:
# P(x < 198) = .99865 (для 3.0 в z-таблице)
# P(x < 150) = .00135 (для -3.0 в z-таблице)
.00135 + 1 - .99865

0.0026999999999999247

In [68]:
# используя библиотеку scipy
stats.norm.cdf(150, loc=174, scale=8) + 1 - stats.norm.cdf(198, loc=174, scale=8)

0.002699796063260096

In [69]:
# 8. P(x < 166) = .15866 (для -1.0 в z-таблице)
# /////////// 
# ----------|-----------
#          166 

In [70]:
# используя библиотеку scipy
stats.norm.cdf(166, loc=174, scale=8)

0.15865525393145707

Ответ:

1) P(x > 182) = 0.15866

2) P(x > 190) = 0.02275

3) P(166 < x < 190) = 0.81859

4) P(166 < x < 182) = 0.68268

5) P(158 < x < 190) = 0.9545

6) P(x < 150) + P(x > 190) = 0.0241

7) P(x < 150) + P(x > 198) = 0.0027

8) P(x < 166) = 0.15866

----------

## Задача 5

На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, 

от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

### Решение

std(X) = 5

Z = (X - M(X)) / 5

In [71]:
(190 - 178) / 5

2.4

Ответ: на 2.4 сигмы.